File: src/main/java/j/examples/rules/combined/PdfExtensionFileRule.java
package j.examples.rules.combined;

import j.examples.validation.ValidationResult;

import java.io.File;

public class PdfExtensionFileRule {

    public ValidationResult validate(File file) {
        String fileExtension = file.getName().substring(file.getName().lastIndexOf('.') + 1);

        if (!fileExtension.equals("pdf")) {
            return ValidationResult.failure("The file %s is not a .pdf file".formatted(file));
        }

        return ValidationResult.success();
    }
}

File: src/main/kotlin/k/examples/rules/combined/PdfExtensionFileRule.kt
package k.examples.rules.combined

import k.examples.validation.ValidationResult
import java.io.File

class PdfExtensionFileRule {
    fun validate(file: File):ValidationResult {
        if (file.extension != "pdf") {
            return ValidationResult
                .failure("The file ${file} is not a .pdf file")
        }

        return ValidationResult.success()
    }
}